### DISCUSSION TITLE
How to trim the white space in every line's end?

### DISCUSSION THREAD

**Discussion by dlintw** [helix-editor/helix#9027] [Q&A] [answered]
I wish to trim every line, by `%_`    not work, because it only trim the  last line.
Q1: Any key is used for that?
Q2: If I want to trim all lines on some file type/language, how to do it?

**Comment by David-Else** [comment]
There are many clumsy workarounds here: https://github.com/helix-editor/helix/issues/1481 or you can wait (hopefully not too long) until: https://github.com/helix-editor/helix/pull/8366

**Comment by gabydd** [comment]
``%s\s+$<ret>d`` is not that bad as a work around to remove all trailing whitespace in my opinion

**Comment by dlintw** [comment]
> `%s\s+$<ret>d` is not that bad as a work around to remove all trailing whitespace in my opinion

This method will remove all my blank lines.  That's not what I want to do.  I wish to keep blank lines.

**Reply by gabydd** [reply]
ah yes you are right ``%<space>+$<ret>d`` is probably better assuming you don't care about tabs

**Reply by dlintw** [reply]
How to setup it as a hotkey? eg. ctrl+=

**Reply by gabydd** [reply, marked as answer]
see the shell scripts keybinds in #1481 if you want to bind to a hotkey

**Reply by dlintw** [reply]
So, it is impossible to convert manual operation `%s<space>+$<ret>d` to hotkey.

**Reply by gabydd** [reply]
currently yes that requires macro keybinds: https://github.com/helix-editor/helix/issues/1383

**Comment by wolf** [comment]
Was just suggested to me: `%s\s+$<Return>c<Return><Esc>`   I'm going to go try it.

Alas.  Automatic indentation is just one of the factors that make this solution not viable.
